### PR TITLE
fix(query) Configurable record container sizes for topK, bottomK and countValues

### PR DESCRIPTION
**Pull request by amolnayak311**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** 
We use a fixed 4k buffer for record containers 

**New behavior :**

We change the container size 40k or any other configurable value for ``Topk``, ``BottomK`` and ``CountValue`` aggregation

### CODE REVIEW

File: core/src/main/resources/filodb-defaults.conf [hunk 1]
```diff
@@ -324,6 +324,7 @@ filodb {
     parser = "antlr"
     container-size-overrides {
         filodb-query-exec-localpartitionreduceaggregateexec-topbottomk = 40960
+        filodb-query-exec-metadataexec                                 = 65536
```
Comment (vishramachandran) [inline core/src/main/resources/filodb-defaults.conf:327]: Use same 64k override for both ?

File: core/src/main/scala/filodb.core/query/QueryConfig.scala [hunk 2]
```diff
@@ -51,7 +51,11 @@ object QueryConfig {
                                            remoteGrpcEndpoint = None,
                                            enforceResultByteLimit = false,
                                            allowPartialResultsRangeQuery = false,
-                                           allowPartialResultsMetadataQuery = true)
+                                           allowPartialResultsMetadataQuery = true,
+                                           recordContainerOverrides =
+                                             Map("filodb-query-exec-localpartitionreduceaggregateexec-topbottomk"
+                                                     -> 40960,
+                                                  "filodb-query-exec-metadataexec" -> 8192))
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/query/QueryConfig.scala:58]: Shouldn't you be using the `container-size-overrides` config ?

File: query/src/main/scala/filodb/query/exec/ExecPlan.scala [hunk 3]
```diff
@@ -60,7 +60,33 @@ trait ExecPlan extends QueryCommand {
...
    */
-  def maxRecordContainerSize: Int = SerializedRangeVector.MaxContainerSize
+  def maxRecordContainerSize(cfg: QueryConfig): Int = {
+    this match {
+      // If there is a ReduceAggregateExec and aggregation is one of TopK, BottomK or CountValues,
+      // use large record container
+      case ra: ReduceAggregateExec
+                    if ra.aggrOp == AggregationOperator.TopK  ||
+                       ra.aggrOp == AggregationOperator.BottomK ||
+                       ra.aggrOp == AggregationOperator.CountValues   =>
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/exec/ExecPlan.scala:70]: Quantile Aggregation also takes space due to t-digests and should be included

**Review by vishramachandran — APPROVED** [review]
Approved with comments

File: core/src/main/scala/filodb.core/query/QueryConfig.scala (hunk 2 above)
Comment (amolnayak311) [reply core/src/main/scala/filodb.core/query/QueryConfig.scala:58]: Actually this is used for a unit where where construction should fail and when using 64 it doesn't.

**Review by vishramachandran — APPROVED** [review]

**Pull request merged**